### PR TITLE
Doc enhancement on distributed training

### PR DESCRIPTION
**Pull request by bryanyzhu**

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1130-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1130/1/index.html 
Code coverage of this PR:  vs. Master:

**Comment by bryanyzhu** [comment]
@eric-haibin-lin Please help review the updated tutorial, thank you.

File: docs/tutorials/distributed/distributed_slowfast.py [hunk 1]
```diff
@@ -82,26 +63,51 @@
 #
 #     ssh-rsa XXXXXXXXXXXXXXXX node1@ip-123-123-1-123
 #
-# The content in the middle is the actuall ssh key, and the IP address in the end is the node's internal IP address.
+# The content in the middle is the actuall ssh key for node1, and the IP address in the end is the node1's internal IP address.
+# Similarly, ssh into other machines (node2, node3 and node4) and perform the same steps to generate the ``id_rsa.pub`` file for each machine.
```
Comment (eric-haibin-lin) [inline docs/tutorials/distributed/distributed_slowfast.py]: what about:
- generate `id_rsa` and `id_rsa.pub` once on node1
- copy both file to all nodes
- for each of the node, append authorized_keys with content of `id_rsa.pub`
?

File: docs/tutorials/distributed/distributed_slowfast.py [hunk 2]
```diff
@@ -42,66 +42,59 @@
 # How to use our code to train a SlowFast model in a distributed manner?
 # ----------------------------------------------------------------------
 #
-# In order to perform distributed training, you need to install MXNet and prepare the cluster ready.
-# MXNet provides a script ``tools/launch.py`` to make it easy to launch distributed training on a cluster with ssh, mpi, sge or yarn.
+# In order to perform distributed training, you need to (1) prepare the cluster ready; (2) install MXNet; and (3) prepare your code
```
Comment (eric-haibin-lin) [inline docs/tutorials/distributed/distributed_slowfast.py]: shall we remove `ready`? prepare already means "make sth ready"

File: docs/tutorials/distributed/distributed_slowfast.py [hunk 3]
```diff
@@ -42,66 +42,59 @@
...
+#     ssh-keygen -t rsa
 #
-# For more installation options (i.e., different versions of MXNet or CUDA),
-# please check `GluonCV installation guide <https://gluon-cv.mxnet.io/install/install-more.html>`_ for more information.
+# Just follow the default, you will have a file named ``id_rsa`` and a file named ``id_rsa.pub``, both under the ``~/.ssh/`` folder.
+# ``id_rsa`` is the private RSA key, and ``id_rsa.pub`` is its public key.
 
+################################################################
+# Second, copy both files (``id_rsa`` and ``id_rsa.pub``) of node1 to all other machines.
+# For each machine, you will fine an ``authorized_keys`` file under ``~/.ssh/`` folder as well.
```
Comment (eric-haibin-lin) [inline docs/tutorials/distributed/distributed_slowfast.py]: `fine` -> `find`. 
It exists on ec2 but might be missing sometimes. Shall we also add 
Q2: I didn't find the file `.ssh/authorized_keys`
Answer: just create one with 
```
touch ~/.ssh/authorized_keys
chmod 600 ~/.ssh/authorized_keys
```

File: docs/tutorials/distributed/distributed_slowfast.py [hunk 4]
```diff
@@ -42,66 +42,59 @@
...
-#     pip install mxnet-cu100
+#     ssh-keygen -t rsa
 #
-# For more installation options (i.e., different versions of MXNet or CUDA),
-# please check `GluonCV installation guide <https://gluon-cv.mxnet.io/install/install-more.html>`_ for more information.
+# Just follow the default, you will have a file named ``id_rsa`` and a file named ``id_rsa.pub``, both under the ``~/.ssh/`` folder.
+# ``id_rsa`` is the private RSA key, and ``id_rsa.pub`` is its public key.
 
+################################################################
+# Second, copy both files (``id_rsa`` and ``id_rsa.pub``) of node1 to all other machines.
```
Comment (eric-haibin-lin) [inline docs/tutorials/distributed/distributed_slowfast.py]: Please make sure the permission of id_rsa is good via `chmod 400 id_rsa` after the copy. 

We can add a FAQ section at the end of this tutorial. For example, 
Q1: I see the error message below, what should I do?
```
Permissions for 'id_rsa' are too open.
It is recommended that your private key files are NOT accessible by others.
This private key will be ignored.
```
Answer: you need to make sure the permission of id_rsa is good via `chmod 400 id_rsa` after the copy.

Comment (bryanyzhu) [reply docs/tutorials/distributed/distributed_slowfast.py]: Thanks a lot, these comments are awesome.

**Comment by mli** [comment]
Job PR-1130-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1130/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1130-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1130/3/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1130-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1130/4/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**